src/renderer/views/Playlist/Playlist.js

```diff
@@ -544,9 +550,14 @@ export default defineComponent({
       }
     },
 
-    setPlaylistTitle: function (value) {
-      this.playlistTitle = value
-      document.title = `${value} - ${packageDetails.productName}`
+    updatePageTitle() {
+      const playlistTitle = this.playlistTitle
+      const channelName = this.channelName
+      const titleText = [
+        playlistTitle,
+        channelName,
+      ].filter(v => v).join(' | ')
```

Comment:
What a clever snippet!

Reply:
JS version of ruby code like (not exact coz [`Array#compact`](https://rubyapi.org/3.3/o/array#method-i-compact) only removes `nil` values)
```ruby
[
  value_a if condition_a,
  value_b if condition_b,
].compact.join(' | ')
```